RELEASES.md

```diff
@@ -0,0 +1,250 @@
+
+The off-chain test environment has lost a bit of power compared to the old ink! lang. It currently is no longer possible to query and set special test data about the environment but these restirctions are going to be lifted in the future.
+
+<table>
+<tr>
+<th>Feature</th>
+<th>v1.0 Syntax</th>
+<th>v2.0 Syntax</th>
+</tr>
+<tr>
+<td>
+Declaring a Contract
+</td>
+<td>
+
+We used to use a single `contract!` macro:
+
+```rust
+contract! {
+	...
+}
+```
+</td>
+<td>
+
+We now use `#[ink::contract(..)]` attribute macros on Rust modules:
+
+```rust
+#[ink::contract(version = "0.1.0")]
+mod erc20 {
+	...
+}
+```
+
+> Note, we now require a mandatory ink! version in the header.
+</td>
+</tr>
+<tr>
+<td>
+Defining Types
+</td>
+<td>
+We used to define types using a special tag like this:
+
+```rust
+#![env = DefaultSrmlTypes]
+```
+</td>
+<td>
+
+This is now automatically defined in your `ink::contract` attribute, but can also be custom defined like:
+
+```rust
+#[ink::contract(version = "0.1.0", )]
+```
+
+TODO
+
+You still have access to the custom types throughout your contract just as before.
+
+</td>
+</tr>
+<tr>
+<td>
+Declaring Storage
+</td>
+<td>
+
+We defined storage items in a single `struct`:
+
+```rust
+struct Erc20 {
+	total_supply: storage::Value<Balance>,
+	balances: storage::HashMap<AccountId, Balance>,
+	allowances: storage::HashMap<(AccountId, AccountId), Balance>,
+}
+```
+
+</td>
+<td>
+
+We just need to add the attribute `ink(storage)`:
+
+```rust
+#[ink(storage)]  // <-- This
+struct Erc20 {
+	total_supply: storage::Value<Balance>,
+	balances: storage::HashMap<AccountId, Balance>,
+	allowances: storage::HashMap<(AccountId, AccountId), Balance>,
+}
+```
+</td>
+</tr>
+
+<tr>
+<td>
+Defining a Constructor
+</td>
+<td>
+
+We used to define our constructor by implementing `Deploy` and the `deploy` function:
+
+```rust
+impl Deploy for Erc20 {
+	fn deploy(&mut self, init_supply: Balance) {
+		let caller = env.caller();
+		self.total_supply.set(init_value);
+		self.balances.insert(caller, init_supply);
+		env.emit(Transfer {
+			from: None,
+			to: Some(env.caller()),
+			value: init_value
+		});
+	}
+}
+```
+</td>
+<td>
+We can now create multiple different constructors by simply creating functions with the correct tag:
+
+```rust
+impl Erc20 {
+	#[ink(constructor)]
+	fn new(&mut self, initial_supply: Balance) {
+		let caller = self.env().caller();
+		self.total_supply.set(initial_supply);
+		self.balances.insert(caller, initial_supply);
+		self.env().emit_event(Transferred {
+			from: None,
+			to: Some(caller),
+			amount: initial_supply,
+		});
+	}
+//-snip-
+```
+
+You can name them whatever you want.
```

Comment:
The prefix `__ink` for functions is reserved, if a constructor is named that way the contract won't compile.